e2etests/web/regular_integration_tests/test_driver/text_editing_integration.dart

```diff
@@ -243,8 +245,8 @@ void main() {
     expect(input.hasAttribute('readonly'), isTrue);
 
     // Make sure the entire text is selected.
-    TextRange range =
-        TextRange(start: input.selectionStart, end: input.selectionEnd);
+    TextRange range = TextRange(
+        start: input.selectionStart ?? 0, end: input.selectionEnd ?? 0);
```

Comment:
`??` is essentially an `if`/`else`, which means two paths are possible, one when `input.selectionStart` is null and another when it isn't. However, a test is only executed once, which means only one situation should be possible, and the other situation is likely a failure of some sort.

Reply:
I think it does not change the result for one this particular method is covering. This is an integration tests, the other conditions related to text selection ranges are covered in the unit tests.

The purpose of the test is to see if the text in the selection range is 'Lorem'.  Whether `input.end` being null or 0 will give the same outcome, since the selection range won't be 'Lorem' in either case, this method should fail.

Reply:
Is `input` being `null` semantically equivalent to `input.end` being 0? Or would it indicate a bug in the framework? If it's the latter, then failing due to "Lorem" string not being selected would be misleading because the actual bug is that the framework sent `null`, which was not expected. If they both mean the same thing, then I guess it's OK to fallback to zeros. It still feels strange to me that we don't know what the framework will send.